http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -1365,11 +1364,17 @@ class LeoPubsubMessageSubscriber[F[_]](
                       persistentDiskOpt <- rc.persistentDiskId.flatTraverse(did =>
                         persistentDiskQuery.getPersistentDiskRecord(did).transaction
                       )
-                      _ <- persistentDiskOpt.traverse_(d =>
-                        googleDiskService.deleteDisk(googleProject, rc.zone, d.name) >> persistentDiskQuery
-                          .updateStatus(d.id, DiskStatus.Deleted, now)
-                          .transaction
-                      )
+                      _ = persistentDiskOpt match {
+                        case Some(value) =>
+                          if (value.status == DiskStatus.Creating || value.status == DiskStatus.Failed) {
```

Comment:
Huh... I'm surprised this compiles?

This is probably the reason your test is failing. You never actually run the IO from `persistentDiskOpt.traverse_(d =>
                              googleDiskService.deleteDisk(googleProject, rc.zone, d.name) >> persistentDiskQuery
                                .updateStatus(d.id, DiskStatus.Deleted, now)
                                .transaction
                            )`

Line 1367 should read 
`_ = persistentDiskOpt match {`

Furthermore, that will cause a syntax error, as the type from the `if() { ... }` branch is IO[_] and the type of the else branch is None (and not included)